### PR TITLE
@Hurumap ChartTooltip

### LINKED ISSUE #776: @hurumap/core/ChartTooltip

### PR DESCRIPTION
**Pull request by koechkevin**
## Description

Moves ChartTooltip component to Hurumap core.

NB: couldn't get `sx` or any other Material UI specific feature to work for Tooltip inside `hurumap/core` except style  + useTheme() properties.

Fixes #776

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/pesayetu/src/components/HURUmap/Chart/index.js [hunk 1]
```diff
@@ -101,6 +101,7 @@ function Chart({
           </ThemeProvider>
         </StyledEngineProvider>,
       );
+      console.log(el);
```
Comment (kilemensi) [inline apps/pesayetu/src/components/HURUmap/Chart/index.js:104]:
```suggestion
```

**Review by kilemensi — APPROVED** [review]

**Comment by kilemensi** [comment]
>NB: couldn't get sx or any other Material UI specific feature to work for Tooltip inside hurumap/core except style + useTheme() properties.
>

Didn't see the above comment when I was review @koechkevin. What issues did you run into?

**Comment by koechkevin** [comment]
> > NB: couldn't get sx or any other Material UI specific feature to work for Tooltip inside hurumap/core except style + useTheme() properties.
> 
> Didn't see the above comment when I was review @koechkevin. What issues did you run into?

@kilemensi sx styles just don't get applied when you use inside hurumap-core package.

**Comment by kilemensi** [comment]
Not sure I get you @koechkevin ... but let me pull and run locally.

**Comment by kilemensi** [comment]
I think style work because `style` have higher priority that classes (`sx`). For `sx` to work, we'll need to figure out how to use [custom theme](https://github.com/vega/vega-tooltip/blob/main/docs/customizing_your_tooltip.md).

**Comment by koechkevin** [comment]
I think we'll go with inline style here because if we were to use custom theme as [specified here](https://github.com/vega/vega-tooltip/blob/main/docs/customizing_your_tooltip.md),  we'd have to write custom CSS. Since we may need to use theme properties, it would mean having to write Internal CSS in a JS file which is not ideal.  Style prop is the closest thing in terms of usage to sx in my view.

**Comment by kilemensi** [comment]
> I think we'll go with inline ...
>

You're the closest to the code @koechkevin but **_If_** I understood that doc, no, we wouldn't need to create custom CSS file. `sx` should be all we need.
```jsx
    <ChartOrChartParentComponent
      ...
      sx={{
        "& .vg-tooltip.custom-theme": {
           color: "red",
        }
      }}
      >
```

`& .vg-tooltip.custom-theme` can be `& #vg-tooltip-element.vg-tooltip.custom-theme` if we know/can set id of `Tooltip` element.

But, at the moment, job one is to know, does it work?

**Comment by koechkevin** [comment]
> > I think we'll go with inline ...
> 
> You're the closest to the code @koechkevin but **_If_** I understood that doc, no, we wouldn't need to create custom CSS file. `sx` should be all we need.
> 
> ```js
>     <ChartOrChartParentComponent
>       ...
>       sx={{
>         "& .vg-tooltip.custom-theme": {
>            color: "red",
>         }
>       }}
>       >
> ```
> 
> `& .vg-tooltip.custom-theme` can be `& #vg-tooltip-element.vg-tooltip.custom-theme` if we know/can set id of `Tooltip` element.
> 
> But, at the moment, job one is to know, does it work?

This wouldn't work on the tooltip @kilemensi the parent component of Tooltip is a div which doesn't support sx. Also, its children are  React components which would have to accept sx props independently

**Comment by kilemensi** [comment]
... it doesn't need to be a direct parent @koechkevin . `& .class` would work on any parent; `&>.class`only works on direct parent (if I still remember my CSS correctly)

**Comment by koechkevin** [comment]
@kelvinkipruto @kilemensi I was able to get `sx` to work with vega tooltip. Please have a re look at this PR

File: packages/hurumap-core/src/ChartTooltip/Tooltip.js
Comment (kelvinkipruto) [inline packages/hurumap-core/src/ChartTooltip/Tooltip.js:1]: @koechkevin Looks good. Shouldn't there be tests for `Tooltip`

**Comment by kilemensi** [comment]
> @kelvinkipruto @kilemensi I was able to get `sx` to work with vega tooltip. Please have a re look at this PR

That's the thinking/tenacity I am about!!

**Review by kilemensi — COMMENTED** [review]
👍🏽 

----

Another approach that we could have taken using the recommended [custom theme](https://github.com/vega/vega-tooltip/blob/main/docs/customizing_your_tooltip.md) is via MUI's [Global CSS Override](https://mui.com/material-ui/customization/how-to-customize/#4-global-css-override). It can style even elements in a portal. We wouldn't even need a standalone Tooltip component since we'll just be styling the Vega-Tooltip itself.

File: apps/climatemappedafrica/src/components/HURUmap/Chart/index.js [hunk 2]
```diff
@@ -28,7 +24,7 @@ const useStyles = makeStyles(() => ({
 function Chart({
   indicator,
   indicatorTitle,
-  secondaryIndicator: { indicator: secondaryIndicator },
+  secondaryIndicator: sI,
```
[the remark this answers is not in the document]
Comment (koechkevin) [reply apps/climatemappedafrica/src/components/HURUmap/Chart/index.js:27]: This was a workaround for an error I encountered @kilemensi `Cannot read property indicator of undefined`

**Comment by koechkevin** [comment]
> 👍🏽
> 
> Another approach that we could have taken using the recommended [custom theme](https://github.com/vega/vega-tooltip/blob/main/docs/customizing_your_tooltip.md) is via MUI's [Global CSS Override](https://mui.com/material-ui/customization/how-to-customize/#4-global-css-override). It can style even elements in a portal. We wouldn't even need a standalone Tooltip component since we'll just be styling the Vega-Tooltip itself.

This could work as well @kilemensi but I'd love to have this approach tried to see if there are challenges that would necessitate it. it'd also mean a major change that would best be implemented in a new iteration/version.

**Pull request merged**